### PR TITLE
Add an interface to the match_main method in diff-match-patch.

### PR DESCRIPTION
**Pull request by freakboy3742**

### CODE REVIEW

**Comment by freakboy3742** [comment]
The tests on this patch depend on #7.

File: interface.cpp [hunk 1]
```diff
@@ -46,6 +47,16 @@ struct call_traits<'y'> {
     static PyObject* from_string(std::string& value) {
         return PyString_FromStringAndSize(value.data(), value.size());
     }
+
+    // Convert std::strings to char*s
+    static const char* to_bytes(std::string& value) {
+        return value.c_str();
+    }
+
+    // Convert std::strings to char*s
```
Comment (JoshData) [inline interface.cpp]: Duplicate?

File: interface.cpp [hunk 2]
```diff
@@ -110,6 +121,74 @@ struct call_traits<'u'> {
         free(buf);
         return ret;
     }
+
+/* Python 3.5 introduced Py_EncodeLocale for converting wchar_t* to char*.
+   Backport that method for the benefit of Python 2.7 and 3.4. However,
+   if we're on Python 3.5+, just use the native method. */
+#if PY_MAJOR_VERSION == 2 || (PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION <= 4)
+    static const char* to_bytes(std::wstring& value) {
```
Comment (JoshData) [inline interface.cpp]: Is to_bytes just for raising the RuntimeError?

File: interface.cpp (hunk 1 above)
Comment (freakboy3742) [reply interface.cpp]: Yes. Deleted.

File: interface.cpp (hunk 2 above)
Comment (freakboy3742) [reply interface.cpp]: Correct.

Comment (JoshData) [reply interface.cpp]: The RuntimeError looks like it only occurs in very unexpected conditions, right? I'm just thinking that this is a lot of code for error handling that may never get executed.

**Comment by freakboy3742** [comment]
@JoshData Well sure - but the alternative is an exception that isn't helpful when it *is* raised.

Well..., there's one other option - drop support for Python 2.7 and 3.4. Python 3.5 has `Py_EncodeLocale` available natively, which is all that is really needed.

**Comment by JoshData** [comment]
I'm fine dropping support for nice error messages in 2.7 and 3.4. :) Can just replace the body of the function with `return "unspecified error"` :).

**Comment by JoshData** [comment]
Thanks!

**Pull request merged**